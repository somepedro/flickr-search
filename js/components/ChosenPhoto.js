import React from 'react'
import { object, func } from 'prop-types'
import { buildPhotoUrl } from '../helpers'

class Header extends React.Component {
  render () {
    const { photo, onNavigateLeft, onNavigateRight } = this.props
    if (!photo) {
      return null
    } else {
      return (
        <div className='chosen-photo-cover'>
          <div className='chosen-photo-container'>
            <div className='arrow left-arrow' onClick={onNavigateLeft} />
            <div className='arrow right-arrow' onClick={onNavigateRight} />
            <div className='chosen-photo' style={{backgroundImage: `url(${buildPhotoUrl(photo)})`}} title={photo.title} />
          </div>
        </div>
      )
    }
  }
}

Header.propTypes = {
  photo: object,
  onNavigateLeft: func.isRequired,
  onNavigateRight: func.isRequired
}

export default Header
