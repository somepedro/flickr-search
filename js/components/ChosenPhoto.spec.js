/* global test, expect */
import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import ChosenPhoto from './ChosenPhoto'

test('ChosenPhoto snapshot test', () => {
  const component = shallow(<ChosenPhoto onNavigateLeft={() => {}} onNavigateRight={() => {}} />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
