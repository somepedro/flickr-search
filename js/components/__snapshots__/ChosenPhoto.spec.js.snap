// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChosenPhoto snapshot test 1`] = `null`;
